trilitech/jstz
# Infra: `jstz-rollup`

# Context






We need an easy way of deploying and running the `jstz` rollup in CD + Docker. This is more involved than initially thought since we require custom installer kernel configs generated by Rust (since we rely on `bincode` for these configs). 

# Description




**Dependencies**: #265 

This PR:
- Moves the existing logic for deploying and running a rollup from the `jstz_cli` to `jstz_rollup`.
- Creates a CLI tool `jstz-rollup` capable of running and deploying the `jstz` rollup. 
- Updates the `jstz` CLI to use `jstz_rollup` as a crate. 


# Manually testing the PR



Try running the sandbox to ensure these changes don't affect it. 

Additionally I recommend trying out all the commands in `jstz-rollup` in the following order:
- `operator gen-keys`
- `operator get-info`
- `make-installer`
- `deploy-ctez`
- `deploy-bridge`
- `deploy`
- `run`

To configure `jstz-rollup`, pass a `config.json` file in using `--config`.

Example config:
```
{
    "octez_client_dir": "/Users/ajob410/.jstz/rollup/.octez-client",
    "octez_node_endpoint": "https://rpc.nairobinet.teztnets.xyz/",
    "octez_rollup_node_dir": "/Users/ajob410/.jstz/rollup/.smart-rollup-node"
}
```  

Also make sure you have octez installed.
To obtain funds for running and deploying `jstz`, use: https://faucet.nairobinet.teztnets.xyz/
